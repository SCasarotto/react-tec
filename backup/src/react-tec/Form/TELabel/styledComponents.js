import styled from 'styled-components'

export const Label = styled.label`
    ${(props) => {
        const { disabled, htmlFor, theme } = props

        let cursor = 'default'
        if (htmlFor) {
            cursor = 'pointer'
        }
        if (disabled) {
            cursor = 'not-allowed'
        }

        return `
            position: relative;
            display: block;
            width: auto;
            box-sizing: border-box;
            margin-bottom: 3px;
            font-weight: 100;
            letter-spacing: 0.5px;
            font-size: 18px;
            color: ${disabled ? theme.gray : theme.darkerGray};
            cursor: ${cursor};

            @media (max-width: 550px) {
                font-size: 16px;
            }
        `
    }}
`
