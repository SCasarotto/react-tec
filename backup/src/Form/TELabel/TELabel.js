import React from 'react'
import PropTypes from 'prop-types'

import { Label } from './styledComponents'

const TELabel = (props) => {
	const { children, required, className = '', ...rest } = props
	return (
		<Label className={`TELabel ${className}`} {...rest}>
			{children}
			{required ? ' *' : ''}
		</Label>
	)
}

TELabel.propTypes = {
	htmlFor: PropTypes.string,
	required: PropTypes.bool,
	disabled: PropTypes.bool,
}

export default TELabel
