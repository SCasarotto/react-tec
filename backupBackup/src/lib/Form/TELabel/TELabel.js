//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Label } from './styledComponents'

const TELabel = (props) => {
    const { children, required, ...rest } = props
    return (
        <Label {...rest}>
            {children}
            {required ? ' *' : ''}
        </Label>
    )
}

TELabel.propTypes = {
    labelText: PropTypes.string,
    style: PropTypes.object,
    htmlFor: PropTypes.string,
}

export default TELabel
