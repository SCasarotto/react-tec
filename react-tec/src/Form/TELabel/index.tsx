import React from 'react'

import { Label } from './styledComponents'

export interface TELabelCustomProps {
	required?: boolean
	disabled?: boolean
}
export interface TELabelProps
	extends React.DetailedHTMLProps<
			React.LabelHTMLAttributes<HTMLLabelElement>,
			HTMLLabelElement
		>,
		TELabelCustomProps {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}

export const TELabel: React.FC<TELabelProps> = (props) => {
	const { children, required, className = '', ...rest } = props
	return (
		<Label className={`TELabel ${className}`} {...rest}>
			{children}
			{required ? ' *' : ''}
		</Label>
	)
}
