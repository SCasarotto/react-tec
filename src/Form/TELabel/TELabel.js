import React from 'react'
import PropTypes from 'prop-types'

import { Label } from './styledComponents'

const TELabel = (props) => {
	const { children, required, ...rest } = props
	return (
		<Label {...rest}>
			{children}
			{required ? ' *' : ''}
		</Label>
	)
}

TELabel.propTypes = {
	htmlFor: PropTypes.string,
	required: PropTypes.bool,
	disabled: PropTypes.bool,
}

export default TELabel
