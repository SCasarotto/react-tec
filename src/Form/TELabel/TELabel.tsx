import React from 'react'

import { Label } from './styledComponents'

// TODO: See if these can be merged and not separate ... seems weird
export interface TELabelCustomProps {
	required?: boolean
	disabled?: boolean
}
export interface TELabelProps
	extends React.LabelHTMLAttributes<HTMLLabelElement>,
		TELabelCustomProps {}
/**
 *
 * A simple styling of an html `<label>`.
 *
 * Besides the props mentioned below, all props are passed right through to the label. Reference the [HTML documentation for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label).
 *
 */
export const TELabel = React.forwardRef<HTMLLabelElement, TELabelProps>((props, ref) => {
	const { children, required, className = '', ...rest } = props
	return (
		<Label className={`TELabel ${className}`} ref={ref} {...rest}>
			{children}
			{required ? ' *' : ''}
		</Label>
	)
})
