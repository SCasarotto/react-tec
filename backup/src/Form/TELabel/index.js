import TELabel from './TELabel'

export default TELabel
