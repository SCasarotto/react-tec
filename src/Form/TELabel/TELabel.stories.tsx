import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TELabel, TELabelProps } from './TELabel'

export default {
	title: 'Form/TELabel',
	component: TELabel,
	argTypes: {},
} as Meta

const Template: Story<TELabelProps> = (args) => <TELabel {...args} />

export const Default = Template.bind({})
Default.args = {
	children: 'Default',
}

export const Required = Template.bind({})
Required.args = {
	children: 'Required',
	required: true,
}

export const Disabled = Template.bind({})
Disabled.args = {
	children: 'Disabled',
	disabled: true,
}
