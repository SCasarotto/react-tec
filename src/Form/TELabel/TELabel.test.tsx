import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { TELabel } from './TELabel'

describe('TELabel', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TELabel />)
		expect(asFragment()).toMatchSnapshot()
	})

	// Class Name
	it('accepts custom className', () => {
		const { container } = render(<TELabel className='myClassName' />)
		expect(container.firstChild).toHaveClass('myClassName')
	})

	// Ref
	it('renders with ref', async () => {
		const ref: { current: HTMLLabelElement | null } = { current: null }
		render(<TELabel ref={ref} data-testid='myLabel' />)
		// Using this as the focus check didn't seem to work
		waitFor(() => expect(ref.current).toBeTruthy())
	})
})
