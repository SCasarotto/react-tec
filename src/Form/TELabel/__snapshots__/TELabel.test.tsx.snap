// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TELabel matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: block;
  width: auto;
  box-sizing: border-box;
  margin-bottom: 3px;
  font-size: 18px;
  cursor: default;
}

@media (max-width:550px) {
  .c0 {
    font-size: 16px;
  }
}

<label
    class="c0 TELabel "
  />
</DocumentFragment>
`;
